### ISSUE TITLE
Unable to create a WAF certificate with a Let's Encrypt certificate

### ISSUE THREAD

**Issue by hcantzler** [opentelekomcloud/terraform-provider-opentelekomcloud#1090]
The problem is that I can't create a WAF certificate from Let's Encrypt certificate files. I am able to use the same certificate files with opentelekomcloud_lb_certificate_v2 without a problem. I am also able to create the WAF certificate manually on the portal with the certificate content. Thus, I suppose the certificate itself is fine.

# Terraform Version
Terraform v0.14.6
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/opentelekomcloud/opentelekomcloud v1.23.12

# Affected Resource(s)
opentelekomcloud_waf_certificate_v1

# Terraform Configuration Files
```hcl
resource "opentelekomcloud_waf_certificate_v1" "appfw_certificate" {
  name        = "dLS-${var.env}-appfw-certificate"
  key = file("./dev.tsi-poc.de/privkey1.pem")
  content = file("./dev.tsi-poc.de/cert1.pem")
}
```

# Debug Output/Panic Output
```
Error: Error creating OpenTelekomcomCloud WAF Certificate: Bad request with: [POST https://waf.eu-de.otc.t-systems.com/v1/4bcc562f125740fbba534078321c1406/waf/certificate], error message: {"error_msg": "The certificate is invalid","error_code":"WAF.1605"}

  on app_fw.tf line 5, in resource "opentelekomcloud_waf_certificate_v1" "appfw_certificate":
   5: resource "opentelekomcloud_waf_certificate_v1" "appfw_certificate" {
```

# Steps to Reproduce
1. Create a Let's Encrypt certificate (`certbot certonly --manual --preferred-challenges dns --rsa-key-size 4096 -d your.domain
`)
2. `terraform apply`

# Expected Behavior
I would expect that the WAF certificate is created assuming that I provide valid private key and certificate.

# Actual Behavior

# Important Factoids

# References

**Comment by outcatcher** [comment]
For some reason, WAF requires removing `\n` from the certificate and the key. This implicitly done in the documentation, but I will add a note for that.

Following configuration works for me:
```hcl
resource "opentelekomcloud_waf_certificate_v1" "appfw_certificate" {
  name    = "my-waf-certificate"
  key     = replace(file("~/tmpcert/key.pem"), "\n", "")
  content = replace(file("~/tmpcert/cert.pem"), "\n", "")
}
```

**Comment by hcantzler** [comment]
Yes, this works for me. There are no deployment erros anymore. Thank you!

**Comment by outcatcher** [comment]
The certificate uploaded this way is broken (at least has no expiration date), please don't use this approach. We will fix this internally.

**Comment by outcatcher** [comment]
After merging the #1098 configuration from your original description should work correctly.

**Comment by outcatcher** [comment]
Basically, everything about `waf_certificate_v1` is working incorrectly in the current release.

**Comment by hcantzler** [comment]
Thank you for the update. I also noticed some communication failure when connecting to the WAF after the successful deployment. Please, give me a quick update when this is working.

**Comment by outcatcher** [comment]
What exactly do you mean by "communication failure"?

As for certificates:
Currently, a successful deployment can be done only without line separators in the cert/key. In this case, the expiration date is not parsed by the service, not read by the terraform (otherwise there is an error). So I wonder if a certificate is really working for the WAF in this case.

**Comment by hcantzler** [comment]
I did a successful deployment using keys without line breaks. Good good, but...

I observed the following "communication failure" when accessing the public WAF endpoint (of the WAF domain). 
curl https://dev.tsi-poc.de/
curl: (35) error:14094438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error

Solution for me was to create the certificate manually with the my keys and configured the WAF domain with the keys. I could then access the WAF endpoint without errors. So, looks like it is really a certificate problem.
